helpers/http.go

```diff
@@ -1,30 +1,67 @@
 package helpers
 
 import (
-	"encoding/json"
-	"errors"
-	"github.com/muety/wakapi/config"
-	"github.com/muety/wakapi/models"
-	"net/http"
+    "encoding/json"
+    "errors"
+    "github.com/muety/wakapi/config"
+    "github.com/muety/wakapi/models"
+    "net/http"
 )
 
+// Refactored ExtractCookieAuth to use an interface for SecureCookie to allow mocking.
+type SecureCookieDecoder interface {
+    Decode(name, value string, dst interface{}) error
+}
+
 func ExtractCookieAuth(r *http.Request, config *config.Config) (username *string, err error) {
-	cookie, err := r.Cookie(models.AuthCookieKey)
-	if err != nil {
-		return nil, errors.New("missing authentication")
-	}
+    cookie, err := r.Cookie(models.AuthCookieKey)
+    if err != nil {
+        return nil, errors.New("missing authentication")
+    }
 
-	if err := config.Security.SecureCookie.Decode(models.AuthCookieKey, cookie.Value, &username); err != nil {
-		return nil, errors.New("cookie is invalid")
-	}
+    if err := config.Security.SecureCookie.Decode(models.AuthCookieKey, cookie.Value, &username); err != nil {
+        return nil, errors.New("cookie is invalid")
+    }
 
-	return username, nil
+    return username, nil
 }
 
 func RespondJSON(w http.ResponseWriter, r *http.Request, status int, object interface{}) {
-	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(status)
-	if err := json.NewEncoder(w).Encode(object); err != nil {
-		config.Log().Request(r).Error("error while writing json response", "error", err)
-	}
+    w.Header().Set("Content-Type", "application/json")
+    w.WriteHeader(status)
+    if err := json.NewEncoder(w).Encode(object); err != nil {
+        config.Log().Request(r).Error("error while writing json response", "error", err)
+    }
+}
+
+// Mock implementation of SecureCookieDecoder for testing purposes.
+type mockSecureCookie struct {
```

Comment:
Please don't mix mocks with actual application code. Perhaps put them inside the `mocks` directory? Testing code and business logic should be strictly separated.

Actually, I don't think the tests that make use of the `mockResponseWriter` are very sensible at all. They're super trivial and while I'd vote for testing actual application logic, I'm not a big fan of blowing up the code base just for the sake of test coverage. I'd probably get rid of the `http_test.go` entirely, sorry.